Detectors/ITSMFT/MFT/tracking/src/TrackExtrap.cxx

```diff
@@ -0,0 +1,276 @@
+  TMatrixD jacob(5, 5);
+  jacob.UnitMatrix();
+  jacob(0, 1) = dZ;
+  jacob(2, 3) = dZ;
+
+  // Extrapolate track parameter covariances to "zEnd"
+  TMatrixD tmp(trackParam->getCovariances(), TMatrixD::kMultTranspose, jacob);
+  TMatrixD tmp2(jacob, TMatrixD::kMult, tmp);
+  trackParam->setCovariances(tmp2);
+
+  // Update the propagator if required
+  if (updatePropagator) {
+    trackParam->updatePropagator(jacob);
+  }
+}
+
+//__________________________________________________________________________
+void TrackExtrap::helixExtrapToZ(TrackParam* trackParam, double zEnd)
+{
+  /// Track parameters extrapolated to the plane at "zEnd" considering a helix
+  /// On return, results from the extrapolation are updated in trackParam.
+
+  if (trackParam->getZ() == zEnd) {
+    return; // nothing to be done if same z
+  }
+
+  // Compute track parameters
+  double dZ = (zEnd - trackParam->getZ()) / 100.; // Propagate in meters
+  double x0 = trackParam->getX();
+  double y0 = trackParam->getY();
+  double phi0 = trackParam->getPhi();
+  double cosphi0 = TMath::Cos(phi0);
+  double sinphi0 = TMath::Sin(phi0);
+  double tanl0 = trackParam->getTanl();
+  double invqpt0 = trackParam->getInvQPt();
+
+  double k = -mBZField * 0.3;
+  double deltax = (dZ * cosphi0 / tanl0 - dZ * dZ * k * invqpt0 * sinphi0 / (2. * tanl0 * tanl0)) * 100;
+  double deltay = (dZ * sinphi0 / tanl0 + dZ * dZ * k * invqpt0 * cosphi0 / (2. * tanl0 * tanl0)) * 100;
+
+  double x = x0 + deltax;
+  double y = y0 + deltay;
+  double deltaphi = +dZ * k * invqpt0 / tanl0;
+  //std::cout << "    Deltaphi extrap = " << deltaphi << " dZ = " << dZ << std::endl;
+  //std::cout << "      Deltax extrap = " << deltax << " = " << (dZ * cosphi0 / tanl0)*100 << " + " << (- dZ * dZ * k * invqpt0 * sinphi0 / (2. * tanl0 * tanl0))*100 << std::endl;
+  //std::cout << "      Deltay extrap = " << deltay << std::endl;
+
+  float phi = phi0 + deltaphi;
+  //o2::utils::BringToPMPi(phi);
+  double tanl = tanl0;
+  double invqpt = invqpt0;
+  trackParam->setX(x);
+  trackParam->setY(y);
+  trackParam->setZ(zEnd);
+  trackParam->setPhi(phi);
+  trackParam->setTanl(tanl);
+  trackParam->setInvQPt(invqpt);
+}
+
+//__________________________________________________________________________
+void TrackExtrap::helixExtrapToZCov(TrackParam* trackParam, double zEnd, bool updatePropagator)
+{
+  // TODO: Implement helix covariance extrapolation
+
+  helixExtrapToZ(trackParam, zEnd);
+
+  // Calculate the jacobian related to the track parameters linear extrapolation to "zEnd"
+  double dZ = (zEnd - trackParam->getZ()) / 100.; // Propagate in meters
+  double phi0 = trackParam->getPhi();
+  double tanl0 = trackParam->getTanl();
+  double invqpt0 = trackParam->getInvQPt();
+  double dZ2 = dZ * dZ;
+  double cosphi0 = TMath::Cos(phi0);
+  double sinphi0 = TMath::Sin(phi0);
+  double tanl0sq = tanl0 * tanl0;
+  double k = -mBZField * 0.3;
+
+  TMatrixD jacob(5, 5);
```

Comment:
idem, you can use here a fixed size matrix, no need for TMatrixD

Reply:
idem